src/backend/distributed/executor/insert_select_executor.c

```diff
@@ -71,8 +75,52 @@ CoordinatorInsertSelectExecScan(CustomScanState *node)
 			LockPartitionRelations(targetRelationId, RowExclusiveLock);
 		}
 
-		ExecuteSelectIntoRelation(targetRelationId, insertTargetList, selectQuery,
-								  executorState);
+		shardConnectionsHash = ExecuteSelectIntoRelation(targetRelationId,
+														 insertTargetList, selectQuery,
+														 executorState,
+														 intermediateResultIdPrefix);
+
+		if (distributedPlan->workerJob != NULL)
```

Comment:
Do you think it's worth writing a short README (<1 page) on how INSERT..SELECT via the coordinator works?

Reply:
Yes, it makes sense. I will update the PR with a README.